src/cluster/detail/ArborX_FDBSCAN.hpp

```diff
@@ -34,7 +34,7 @@ struct CountUpToN
   KOKKOS_FUNCTION auto operator()(Query const &query, Value const &) const
   {
     auto i = getData(query);
-    Kokkos::atomic_increment(&_counts(i));
+    Kokkos::atomic_inc(&_counts(i));
 
     if (_counts(i) < _n)
```

Comment:
This looks super sketchy
```suggestion
    auto const counts_i = Kokkos::atomic_inc_fetch(&_counts(i));

    if (counts_i < _n)
```

Reply:
It's not really sketchy, but you are right, it is could be suboptimal in the case of threads sharing a query (which they currently don't when running BVH). I'll fix it.